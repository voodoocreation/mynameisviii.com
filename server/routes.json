{
  "/appearances/find": "/appearances",
  "/appearances/:slug": "/appearance",
  "/news/find": "/news",
  "/news/:slug": "/newsArticle",
  "/releases/find": "/releases",
  "/releases/:slug": "/release"
}
