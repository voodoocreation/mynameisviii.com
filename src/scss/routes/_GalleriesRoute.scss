.GalleriesRoute {
  padding-bottom: $spacing;
}

.GalleryListings {
  display: flex;
  flex-flow: row wrap;
  margin: (-($spacing / 2));
}
