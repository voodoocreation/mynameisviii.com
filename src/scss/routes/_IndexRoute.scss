.Home {
  margin: (-($spacing / 2));
  padding-bottom: ($spacing / 2);

  h2 {
    @include line-before-after();

    font-size: 14px;
    margin-top: 0;
  }

  > section {
    padding: ($spacing / 2);
  }

  &-news {
    h2 {
      @include hide-visually();
    }
  }

  &-appearances,
  &-bio,
  &-connect {
    > div {
      @include panel-1();

      min-height: 100%;
      padding: $spacing;
    }
  }

  &-appearances-items {
    margin: (-($spacing / 2));
  }

  &-connect {
    &-content {
      display: flex;
      flex-wrap: wrap;

      > * {
        flex-basis: 100%;
        flex-grow: 1;
      }

      h2 {
        align-self: flex-start;
      }
    }

    &-websites {
      display: flex;
      flex-wrap: wrap;
      align-self: center;
      margin: (-($spacing / 2));
      padding: ($spacing / 2);
    }

    &-press {
      align-self: flex-end;
      text-align: center;
    }
  }

  @include screen-below($screen-xxs-max) {
    &-connect-press {
      font-size: 12px;
    }
  }

  @include screen-above($screen-md-min) {
    display: flex;
    flex-wrap: wrap;

    &-bio,
    &-connect {
      flex-basis: 50%;
    }

    &-news,
    &-appearances {
      flex-basis: 100%;
    }
  }

  @include screen-above(1300px) {
    .Home-appearances {
      flex-basis: 30%;

      &-content {
        display: flex;
        flex-direction: column;
      }

      &-items {
        display: flex;
        flex-direction: column;
        flex-grow: 1;

        .AppearanceListing {
          display: flex;
          flex-grow: 1;

          a {
            flex-grow: 1;
          }
        }
      }
    }

    &.hasAppearancesSection {
      .Home-news {
        flex-basis: 70%;
      }
    }
  }
}
