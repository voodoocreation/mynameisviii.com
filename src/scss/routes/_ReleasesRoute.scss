.ReleaseListings {
  margin-bottom: $spacing;
  padding: $spacing;
  background: $panel-color;

  h2 {
    @include line-before-after();

    margin-top: 0;
  }

  &-items {
    display: flex;
    flex-flow: row wrap;
    margin: (-($spacing / 2));
  }
}
