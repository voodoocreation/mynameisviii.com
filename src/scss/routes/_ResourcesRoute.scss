.ResourceListings {
  @include panel-1();

  margin-bottom: $spacing;
  padding: $spacing;

  h2 {
    @include line-before-after();

    margin-top: 0;
  }

  &-items {
    display: flex;
    flex-wrap: wrap;
    margin: (-($spacing / 2));
  }
}
