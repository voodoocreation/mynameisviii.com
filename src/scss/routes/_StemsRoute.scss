.StemsRoute {
  padding-bottom: $spacing;
}

.StemListings {
  display: flex;
  flex-flow: row wrap;
  margin: (-($spacing / 2));
}
