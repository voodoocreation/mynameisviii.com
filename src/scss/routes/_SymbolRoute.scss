.SymbolRoute {
  h2 {
    @include line-before-after();
  }

  section {
    margin-bottom: $spacing;
    padding: ($spacing / 2) $spacing;
    background: $panel-color;
    overflow: hidden;
  }

  &-symbol {
    width: 80%;
    margin: auto;

    &::before {
      @include bg-light();
      @include logo-symbol();

      display: block;
      content: "";
      width: 100%;
      padding-top: 100%;
    }
  }

  &-photo {
    @include square-image();
  }

  @include screen-above($screen-xs-min) {
    &-symbol,
    &-photo {
      float: right;
      width: 30%;
      margin-left: $spacing;
      margin-bottom: $spacing;
    }
  }

  @include screen-above(1500px) {
    &-symbol,
    &-photo {
      width: 20%;
    }
  }
}
