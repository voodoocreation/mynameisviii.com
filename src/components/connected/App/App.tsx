// tslint:disable:no-submodule-imports
import withReduxSaga from "next-redux-saga";
import withRedux from "next-redux-wrapper";
import App, { AppComponentProps, Container } from "next/app";
import { NextDocumentContext as Context } from "next/document";
import * as React from "react";
import { addLocaleData, IntlProvider } from "react-intl";
import { Provider } from "react-redux";
import { Store } from "redux";

import routes from "../../../../next.routes";
import * as actions from "../../../actions/root.actions";
import { isServer } from "../../../helpers/dom";
import * as selectors from "../../../selectors/root.selectors";
import createStore from "../../../store/root.store";

import Shell from "../Shell/Shell";

import en from "react-intl/locale-data/en";
// tslint:enable:no-submodule-imports

addLocaleData([...en]);

type NextPageComponent = React.ComponentType<any> & {
  getInitialProps: (props: any) => any;
};

interface IProps extends AppComponentProps {
  Component: NextPageComponent;
  ctx: Context & { store: any };
  intlProps: {
    locale: string;
    intlMessages: {};
    initialNow: Date;
  };
  pageProps: {};
  store: Store<any>;
}

const getIntlProps = (ctx: Context) => {
  const requestProps = isServer()
    ? ctx.req
    : window.__NEXT_DATA__.props.initialProps.intlProps;
  const { locale, intlMessages } = requestProps;

  return {
    initialNow: Date.now(),
    intlMessages: intlMessages || {},
    locale: locale || "en-NZ"
  };
};

class Application extends App {
  public static async getInitialProps({ ctx, Component }: IProps) {
    let pageProps = {};

    const unsubscribe = ctx.store.subscribe(() => {
      const state = ctx.store.getState();
      const pageError = selectors.getPageError(state);

      if (pageError && ctx.res) {
        ctx.res.statusCode = pageError.status;
        unsubscribe();
      }
    });

    if (Component.getInitialProps) {
      pageProps = await Component.getInitialProps({ ctx });
    }

    const intlProps = getIntlProps(ctx);

    return { pageProps, intlProps };
  }

  private serviceWorkerContainer: any = undefined;

  public componentWillMount() {
    if (
      !isServer() &&
      "serviceWorker" in navigator &&
      (window.location.protocol === "https:" ||
        window.location.hostname === "localhost")
    ) {
      // tslint:disable-next-line
      this.serviceWorkerContainer = require("serviceworker-webpack-plugin/lib/runtime");
    }

    routes.Router.onRouteChangeStart = this.onRouteChangeStart;
    routes.Router.onRouteChangeComplete = this.onRouteChangeComplete;
    routes.Router.onRouteChangeError = this.onRouteChangeError;
  }

  public componentDidMount() {
    const { store } = this.props as IProps;

    store.dispatch(actions.updateOnlineStatus(navigator.onLine));
    store.dispatch(actions.setCurrentRoute(routes.Router.route));

    if (this.serviceWorkerContainer) {
      navigator.serviceWorker.addEventListener(
        "message",
        this.onReceiveServiceWorkerPostMessage
      );

      window.addEventListener(
        "beforeinstallprompt",
        this.onBeforeInstallPrompt
      );

      this.serviceWorkerContainer.register({ scope: "/" }).then(() => {
        window.addEventListener("offline", this.onOnlineStatusChange);
        window.addEventListener("online", this.onOnlineStatusChange);
      });
    }
  }

  public componentWillUnmount() {
    if (this.serviceWorkerContainer) {
      navigator.serviceWorker.removeEventListener(
        "message",
        this.onReceiveServiceWorkerPostMessage
      );

      window.removeEventListener(
        "beforeinstallprompt",
        this.onBeforeInstallPrompt
      );

      window.removeEventListener("offline", this.onOnlineStatusChange);
      window.removeEventListener("online", this.onOnlineStatusChange);
    }
  }

  public render() {
    const { Component, intlProps, pageProps, store } = this.props as IProps;

    return (
      <Container>
        <Provider store={store}>
          <IntlProvider
            initialNow={intlProps.initialNow}
            messages={intlProps.intlMessages}
            locale={intlProps.locale}
          >
            <Shell>
              <Component {...pageProps} />
            </Shell>
          </IntlProvider>
        </Provider>
      </Container>
    );
  }

  private onReceiveServiceWorkerPostMessage = (event: any) => {
    const { store } = this.props as IProps;

    store.dispatch(actions.receiveServiceWorkerMessage(event.data));
  };

  private onBeforeInstallPrompt = async (event: any) => {
    const { store } = this.props as IProps;

    store.dispatch(
      actions.trackEvent({
        event: "addToHomeScreen.prompted"
      })
    );

    const { outcome } = await event.userChoice;

    store.dispatch(
      actions.trackEvent({
        event: "addToHomeScreen.outcome",
        outcome
      })
    );
  };

  private onOnlineStatusChange = (event: Event) => {
    const { store } = this.props as IProps;

    store.dispatch(actions.updateOnlineStatus(event.type === "online"));
  };

  private onRouteChangeStart = (path: string) => {
    const { store } = this.props as IProps;

    store.dispatch(actions.changeRoute.started(path));
  };

  private onRouteChangeComplete = (path: string) => {
    const { store } = this.props as IProps;

    store.dispatch(actions.changeRoute.done({ params: path }));
  };

  private onRouteChangeError = (error: Error, path: string) => {
    const { store } = this.props as IProps;

    store.dispatch(
      actions.changeRoute.failed({
        error: {
          message: error.toString(),
          status: 500
        },
        params: path
      })
    );
  };
}

export default withRedux(createStore)(
  withReduxSaga({ async: true })(Application)
);
