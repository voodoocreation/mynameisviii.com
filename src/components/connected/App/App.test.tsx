import { mount } from "enzyme";
import merge from "lodash.merge";
import * as React from "react";

import App from "./App";

jest.mock("serviceworker-webpack-plugin/lib/runtime", () => ({
  register: jest.fn(() => new Promise(resolve => resolve()))
}));

jest.mock("../../../../next.routes", () => ({
  Router: {
    route: ""
  }
}));

import routes from "../../../../next.routes";

const setup = async (fn: any, fromTestProps?: any) => {
  const Component = () => <div className="PageComponent" />;
  const appProps = merge(
    {
      Component,
      asPath: "",
      ctx: {
        isServer: false,
        pathname: "",
        query: {},
        req: {
          intlMessages: {},
          locale: "en-NZ"
        },
        res: {}
      },
      router: {
        pathname: ""
      }
    },
    fromTestProps
  );
  const initialProps = await App.getInitialProps(appProps);
  const props = {
    ...appProps,
    ...initialProps
  };

  return {
    actual: fn(<App {...props} />),
    props
  };
};

const g: any = global;

describe("[connected] <App />", () => {
  const addEventListener = g.addEventListener;
  const removeEventListener = g.removeEventListener;

  beforeAll(() => {
    g.addEventListener = jest.fn((...args) => addEventListener(...args));
    g.removeEventListener = jest.fn((...args) => removeEventListener(...args));
  });

  beforeEach(() => {
    g.isServer = false;
    g.__NEXT_DATA__ = {
      props: {
        initialProps: {
          intlProps: {}
        }
      }
    };
  });

  afterEach(() => {
    g.__NEXT_DATA__ = undefined;
    jest.clearAllMocks();
  });

  afterAll(() => {
    g.addEventListener = addEventListener;
    g.removeEventListener = removeEventListener;
  });

  it("mounts application correctly on the server", async () => {
    g.isServer = true;
    let isPassing = true;

    try {
      const { actual } = await setup(mount, {
        ctx: { isServer: g.isServer }
      });

      expect(actual.render()).toMatchSnapshot();
      actual.unmount();
    } catch (error) {
      isPassing = false;
    }

    expect(isPassing).toBe(true);
  });

  it("mounts application correctly on the client", async () => {
    let isPassing = true;

    try {
      const { actual } = await setup(mount);

      expect(actual.render()).toMatchSnapshot();
      actual.unmount();
    } catch (error) {
      isPassing = false;
    }

    expect(isPassing).toBe(true);
  });

  it("gets `initialProps` from component correctly", async () => {
    const test = "Test";
    const Component: any = () => <div className="PageComponent" />;
    Component.getInitialProps = async () => ({ test });
    const { actual, props } = await setup(mount, { Component });

    expect(props.initialProps.pageProps).toEqual({ test });

    actual.unmount();
  });

  describe("feature events", () => {
    it("handles single feature correctly", async () => {
      const { actual, props } = await setup(mount);

      window.dispatchEvent(
        new CustomEvent("feature", { detail: "has-test-feature" })
      );

      expect(props.store.getState().features.items).toContain(
        "has-test-feature"
      );

      actual.unmount();
    });

    it("handles multiple features correctly", async () => {
      const { actual, props } = await setup(mount);

      window.dispatchEvent(
        new CustomEvent("feature", {
          detail: ["has-test-feature-1", "has-test-feature-2"]
        })
      );

      expect(props.store.getState().features.items).toContain(
        "has-test-feature-1"
      );
      expect(props.store.getState().features.items).toContain(
        "has-test-feature-2"
      );

      actual.unmount();
    });
  });

  describe("router events", () => {
    it("onRouteChangeStart is handled correctly", async () => {
      const { actual, props } = await setup(mount);

      routes.Router.onRouteChangeStart("/");

      expect(props.store.getState().page.transitioningTo).toBe("/");

      actual.unmount();
    });

    it("onRouteChangeComplete is handled correctly", async () => {
      const { actual, props } = await setup(mount);

      routes.Router.onRouteChangeComplete("/");

      expect(props.store.getState().page.transitioningTo).toBeUndefined();
      expect(props.store.getState().page.currentRoute).toBe("/");

      actual.unmount();
    });

    it("onRouteChangeStart is handled correctly", async () => {
      const { actual, props } = await setup(mount);

      routes.Router.onRouteChangeError(new Error("Server error"), "/");

      expect(props.store.getState().page.transitioningTo).toBeUndefined();
      expect(props.store.getState().page.error).toEqual({
        message: "Error: Server error",
        status: 500
      });

      actual.unmount();
    });
  });

  describe("beforeinstallprompt event", () => {
    it("handles dismissed outcome correctly", async () => {
      let isPassing = true;

      try {
        const { actual } = await setup(mount);

        const event: any = new Event("beforeinstallprompt");
        event.userChoice = new Promise(resolve =>
          resolve({ outcome: "dismissed" })
        );
        await window.dispatchEvent(event);

        actual.unmount();
      } catch (error) {
        isPassing = false;
      }

      expect(isPassing).toBe(true);
    });

    it("handles accepted outcome correctly", async () => {
      let isPassing = true;

      try {
        const { actual } = await setup(mount);

        const event: any = new Event("beforeinstallprompt");
        event.userChoice = new Promise(resolve =>
          resolve({ outcome: "accepted" })
        );
        await window.dispatchEvent(event);

        actual.unmount();
      } catch (error) {
        isPassing = false;
      }

      expect(isPassing).toBe(true);
    });
  });

  describe("service worker", () => {
    it("registers when browser support exists", async () => {
      const { actual } = await setup(mount);
      const { serviceWorkerContainer } = actual
        .childAt(0)
        .childAt(0)
        .instance();

      expect(serviceWorkerContainer).toBeDefined();
      expect(serviceWorkerContainer.register).toHaveBeenCalledWith({
        scope: "/"
      });

      actual.unmount();
    });

    it("updates online status correctly", async () => {
      const { actual, props } = await setup(mount);

      expect(g.findMockCall(g.addEventListener, "offline")).toBeDefined();
      expect(g.findMockCall(g.addEventListener, "online")).toBeDefined();

      g.dispatchEvent(new Event("offline"));
      expect(props.store.getState().page.isOnline).toBe(false);

      g.dispatchEvent(new Event("online"));
      expect(props.store.getState().page.isOnline).toBe(true);

      actual.unmount();
      expect(g.findMockCall(g.removeEventListener, "offline")).toBeDefined();
      expect(g.findMockCall(g.removeEventListener, "online")).toBeDefined();

      window.dispatchEvent(new Event("offline"));
      expect(props.store.getState().page.isOnline).toBe(true);
    });

    it("receives messages from service worker correctly", async () => {
      const { actual, props } = await setup(mount);

      expect(
        g.findMockCall(navigator.serviceWorker.addEventListener, "message")
      ).toBeDefined();

      const event = new MessageEvent("message", {
        data: { type: "serviceWorker.activate" }
      });
      navigator.serviceWorker.dispatchEvent(event);
      expect(props.store.getState().page.hasNewVersion).toBe(true);

      actual.unmount();
      expect(
        g.findMockCall(navigator.serviceWorker.removeEventListener, "message")
      ).toBeDefined();

      navigator.serviceWorker.dispatchEvent(event);
      expect(props.store.getState().page.hasNewVersion).toBe(true);
    });
  });
});
