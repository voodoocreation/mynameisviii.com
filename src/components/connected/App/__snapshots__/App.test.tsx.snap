// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[connected] <App /> when mounting on the client matches snapshot 1`] = `
<article
  class="Page"
>
  <header
    class="Banner"
    role="banner"
  >
    <a
      class="Brand"
      href="/"
    >
      <span>
        My name is Viii: a painter of frequencies and thoughts
      </span>
      <svg />
      <svg />
    </a>
    <nav
      class="Navigation"
      role="navigation"
    >
      <button
        class="Button"
        type="button"
      >
        <span
          class="Button--content"
        >
          Menu
        </span>
      </button>
      <ul>
        <li
          class=""
        >
          <a
            href="/news"
          >
            News
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/releases"
          >
            Releases
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/appearances"
          >
            Appearances
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/galleries"
          >
            Galleries
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/stems"
          >
            Stems
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/resources"
          >
            Resources
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="Page--body"
    role="main"
  >
    <article
      class="ErrorPage"
    >
      <header
        class="PageHeader"
      >
        <h1>
          An error has occurred
        </h1>
      </header>
      <div
        class="ErrorPage--content"
      >
        <svg
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 24 24"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M11 15h2v2h-2zm0-8h2v6h-2zm.99-5C6.47 2 2 6.48 2 12s4.47 10 9.99 10C17.52 22 22 17.52 22 12S17.52 2 11.99 2zM12 20c-4.42 0-8-3.58-8-8s3.58-8 8-8 8 3.58 8 8-3.58 8-8 8z"
          />
        </svg>
        <p>
          An error has occurred
        </p>
      </div>
    </article>
  </main>
  <section
    class="ToastContainer"
  >
    <div
      class="Toast OnlineStatusToast isHidden"
    >
      <div
        class="Toast--content"
      >
        <div
          class="Toast--message"
        >
          You are offline. Some data and external links will be unavailable until you reconnect.
        </div>
        <button
          class="Button Toast--closeButton isStyled"
          title="Close"
          type="button"
        >
          <span
            class="Button--content"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
    <div
      class="Toast NewVersionToast isHidden"
    >
      <div
        class="Toast--content"
      >
        <div
          class="Toast--message"
        >
          <p>
            A new version of the site is available; refresh to see it.
          </p>
          <button
            class="Button NewVersionToast--refreshButton isStyled"
            type="button"
          >
            <span
              class="Button--content"
            >
              Refresh
            </span>
          </button>
        </div>
        <button
          class="Button Toast--closeButton isStyled"
          title="Close"
          type="button"
        >
          <span
            class="Button--content"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
              />
            </svg>
          </span>
        </button>
      </div>
    </div>
  </section>
</article>
`;

exports[`[connected] <App /> when mounting on the server matches snapshot 1`] = `
<article
  class="Page"
>
  <header
    class="Banner"
    role="banner"
  >
    <a
      class="Brand"
      href="/"
    >
      <span>
        My name is Viii: a painter of frequencies and thoughts
      </span>
      <svg />
      <svg />
    </a>
    <nav
      class="Navigation"
      role="navigation"
    >
      <button
        class="Button"
        type="button"
      >
        <span
          class="Button--content"
        >
          Menu
        </span>
      </button>
      <ul>
        <li
          class=""
        >
          <a
            href="/news"
          >
            News
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/releases"
          >
            Releases
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/appearances"
          >
            Appearances
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/galleries"
          >
            Galleries
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/stems"
          >
            Stems
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/resources"
          >
            Resources
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="Page--body"
    role="main"
  >
    <div
      class="MockPageComponent"
    />
  </main>
  <section
    class="ToastContainer"
  />
</article>
`;

exports[`[connected] <App /> when mounting when the page has no getInitialProps method matches snapshot 1`] = `
<article
  class="Page"
>
  <header
    class="Banner"
    role="banner"
  >
    <a
      class="Brand"
      href="/"
    >
      <span>
        My name is Viii: a painter of frequencies and thoughts
      </span>
      <svg />
      <svg />
    </a>
    <nav
      class="Navigation"
      role="navigation"
    >
      <button
        class="Button"
        type="button"
      >
        <span
          class="Button--content"
        >
          Menu
        </span>
      </button>
      <ul>
        <li
          class=""
        >
          <a
            href="/news"
          >
            News
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/releases"
          >
            Releases
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/appearances"
          >
            Appearances
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/galleries"
          >
            Galleries
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/stems"
          >
            Stems
          </a>
        </li>
        <li
          class=""
        >
          <a
            href="/resources"
          >
            Resources
          </a>
        </li>
      </ul>
    </nav>
  </header>
  <main
    class="Page--body"
    role="main"
  >
    <div
      class="MockPageComponent"
    />
  </main>
  <section
    class="ToastContainer"
  />
</article>
`;
